core/src/main/scala/org/apache/spark/util/io/ChunkedByteBuffer.scala

```diff
@@ -222,7 +222,7 @@ private[spark] class ChunkedByteBufferInputStream(
     dispose: Boolean)
   extends InputStream {
 
-  private[this] var chunks = chunkedByteBuffer.getChunks().iterator
+  private[this] var chunks = chunkedByteBuffer.getChunks().filter(_.hasRemaining).iterator
```

Comment:
can you add a comment above, saying that we do this filter because `read` assumes `chunks` has no empty chunk?